core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -118,15 +118,33 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     frameResultBuilder.result
   }
 
+  def generateSuppressCalls(pendingLaunch: Set[PodId], newLaunched: Set[PodId]): List[Call] = {
+    val alreadyLaunchedRoles = newLaunched
+      .map(id => specs.podSpecs.get(id))
```

Comment:
We could remove the optional earlier, this is less allocations for no code complexity cost
```
  .collect { case id if specs.podSpecs.contains(id) => specs.podSpecs(id).role }
```

Reply:
nice, thanks 👍